libsolidity/ast/ASTJsonConverter.cpp

```diff
@@ -323,7 +323,11 @@ bool ASTJsonConverter::visit(FunctionDefinition const& _node)
 {
 	std::vector<pair<string, Json::Value>> attributes = {
 		make_pair("name", _node.name()),
-		make_pair(m_legacy ? "constant" : "isDeclaredConst", _node.stateMutability() == StateMutability::View),
+		make_pair(
+			m_legacy ? "constant" : "isDeclaredConst",
+			_node.stateMutability() == StateMutability::View ||
+			_node.stateMutability() == StateMutability::Pure
```

Comment:
While I still don't agree it belongs to `isDeclaredConstant`, because that refers to the `constant` specifier, I can understand the reasoning for static analyzers not yet aware of the `statemutability` field. Ultimately, removing the obsolete field will solve this.